specifyweb/frontend/js_src/lib/components/DataModel/resource.ts

```diff
@@ -270,7 +270,8 @@ export const getUniqueFields = (model: SpecifyModel): RA<string> =>
     ...Object.entries(businessRuleDefs[model.name]?.uniqueIn ?? {})
       .filter(
         ([_fieldName, uniquenessRules]) =>
-          uniquenessRules in schema.domainLevelIds
+          uniquenessRules in schema.domainLevelIds ||
+          uniquenessRules === undefined
```

Comment:
(I seen https://github.com/specify/specify7/issues/3960#issuecomment-1702819456)

can you double-check that this doesn't mark absolutely every field as unique, and rather only fields that have uniqueness constraints of some sort?

also, if you have a moment to spare, can you update/add the test cases?